addons/project/static/src/views/burndown_chart/burndown_chart_model.js

```diff
@@ -1,14 +1,147 @@
+            this.stageSeq = this.searchParams.context.stage_seq;
+        }
+        return super.load(searchParams);
+    }
+
+    /**
+     * Needed to overwrite whole method, The only change is the sortBy of the temporary datasets (now according to stage sequence)
+     * @protected
+     * @override
+     */
+    _getData(dataPoints) {
+        const { comparisonField, groupBy, mode } = this.metaData;
+
+        let identify = false;
+        if (comparisonField && groupBy.length && groupBy[0].fieldName === comparisonField) {
+            identify = true;
+        }
+        const dateClasses = identify ? this._getDateClasses(dataPoints) : null;
+
+        // dataPoints --> labels
+        let labels = [];
+        const labelMap = {};
+        for (const dataPt of dataPoints) {
+            const x = dataPt.labels.slice(0, mode === "pie" ? undefined : 1);
+            const trueLabel = x.length ? x.join(SEP) : this.env._t("Total");
+            if (dateClasses) {
+                x[0] = dateClasses.classLabel(dataPt.originIndex, x[0]);
+            }
+            const key = JSON.stringify(x);
+            if (labelMap[key] === undefined) {
+                labelMap[key] = labels.length;
+                if (dateClasses) {
+                    if (mode === "pie") {
+                        x[0] = dateClasses.classMembers(x[0]).join(", ");
+                    } else {
+                        x[0] = dateClasses.representative(x[0]);
+                    }
+                }
+                const label = x.length ? x.join(SEP) : this.env._t("Total");
+                labels.push(label);
+            }
+            dataPt.labelIndex = labelMap[key];
+            dataPt.trueLabel = trueLabel;
+        }
+
+        // dataPoints + labels --> datasetsTmp --> datasets
+        const datasetsTmp = {};
+        for (const dataPt of dataPoints) {
+            const { domain, labelIndex, originIndex, trueLabel, value } = dataPt;
+            const datasetLabel = this._getDatasetLabel(dataPt);
+            if (!(datasetLabel in datasetsTmp)) {
+                let dataLength = labels.length;
+                if (mode !== "pie" && dateClasses) {
+                    dataLength = dateClasses.arrayLength(originIndex);
+                }
+                datasetsTmp[datasetLabel] = {
+                    data: new Array(dataLength).fill(0),
+                    trueLabels: labels.slice(0, dataLength), // should be good // check this in case identify = true
+                    domains: new Array(dataLength).fill([]),
+                    label: datasetLabel,
+                    originIndex: originIndex,
+                };
+            }
+            datasetsTmp[datasetLabel].data[labelIndex] = value;
+            datasetsTmp[datasetLabel].domains[labelIndex] = domain;
+            datasetsTmp[datasetLabel].trueLabels[labelIndex] = trueLabel;
+            datasetsTmp[datasetLabel].orderIndex = this.stageSeq[datasetLabel] || 0;
+        }
```

Comment:
We should isolate this part of the code in a method. This way, we'ill just add `orderIndex` to its result.